split_to_CMYK.jsx

```diff
@@ -1 +1,83 @@
                cmyk[1].blendMode = BlendMode.MULTIPLY
                cmyk[2].name = layerColor [1];
                cmyk[2].blendMode = BlendMode.MULTIPLY
                cmyk[3].name = layerColor [0];
                cmyk[3].blendMode = BlendMode.MULTIPLY
              
            }
            
            doc.activeLayer = newLayer;
        }
      } catch (e) { 
    }
}
makeLayers ()

/*function splitCyan () {
	var cyan = doc.activeChannels = [doc.channels.getByName('Cyan')];
	app.executeMenuCommand('selectall');
    
}
splitCyan()*/

/*var cyanChannel = doc.activeChannels = [doc.channels.getByName('Cyan')];
var magChannel = doc.activeChannels = [doc.channels.getByName('Magenta')];
var yelChannel = doc.activeChannels = [doc.channels.getByName('Yellow')];
var blackChannel = doc.activeChannels = [doc.channels.getByName('Black')];*/
\ No newline at end of file
+var doc = app.activeDocument;
+
+//switch color mode to get 4 channels
+var CMYK = doc.changeMode(ChangeMode.CMYK);
+
+//makes 4 layers named after the channels and adjusts blending mode
+function makeLayers() {
+  try {
+    if (app.documents.length) {
+      var layerColor = ["Cyan", "Magenta", "Yellow", "Black"]
+      var doc = activeDocument;
+      var layers = doc.artLayers;
+      var cmyk = [
+        layerA = layers.add(),
+        layerB = layers.add(),
+        layerC = layers.add(),
+        layerD = layers.add()
+      ];
+
+      if (layerColor) {
+        cmyk[0].name = layerColor[3];
+        cmyk[0].blendMode = BlendMode.NORMAL
+        cmyk[1].name = layerColor[2];
+        cmyk[1].blendMode = BlendMode.MULTIPLY
+        cmyk[2].name = layerColor[1];
+        cmyk[2].blendMode = BlendMode.MULTIPLY
+        cmyk[3].name = layerColor[0];
+        cmyk[3].blendMode = BlendMode.MULTIPLY
+
+      }
+
+      doc.activeLayer = newLayer;
+    }
+  } catch (e) {}
+}
+makeLayers()
+
+//copies channels into corresponding layers
+function splitCyan() {
+  doc.activeLayer = doc.layers[4];
+  doc.activeChannels = [doc.channels.getByName('Cyan')];
+  doc.selection.selectAll()
+  doc.selection.copy()
+  doc.activeLayer = doc.layers[0];
+  doc.paste()
+}
+splitCyan()
+
+function splitMagenta() {
+  doc.activeLayer = doc.layers[4];
+  doc.activeChannels = [doc.channels.getByName('Magenta')];
+  doc.selection.selectAll()
+  doc.selection.copy()
+  doc.activeLayer = doc.layers[1];
+  doc.paste()
+}
+splitMagenta()
+
+function splitYellow() {
+  doc.activeLayer = doc.layers[4];
+  doc.activeChannels = [doc.channels.getByName('Yellow')];
+  doc.selection.selectAll()
+  doc.selection.copy()
+  doc.activeLayer = doc.layers[2];
+  doc.paste()
+}
+splitYellow()
+
+function splitBlack() {
+  doc.activeLayer = doc.layers[4];
+  doc.activeChannels = [doc.channels.getByName('Black')];
+  doc.selection.selectAll()
+  doc.selection.copy()
+  doc.activeLayer = doc.layers[3];
+  doc.paste()
+}
+splitBlack()
+
+//sets color mode back to RGB
+var RGB = doc.changeMode(ChangeMode.RGB);
```

Comment:
You should be able to just call `doc.changeMode(...)` here w/out assigning the action to a var, see my comment above for the CMYK color mode change.